src/firmware/app/control/trajectory_planner.h

```diff
@@ -11,293 +13,434 @@ typedef struct FirmwareRobotPathParameters
+ * parameterization trajectory then uses linear interpolation to calculate the equivalent
+ * constant interpolation period trajectory
+ *
+ * NOTE: Constant interpolation period means that the duration of time between each point
+ * on the trajectory is CONSTANT. This is different from a constant parameterization
+ * trajectory which has segments defined at constant polynomial evaluation intervals.
+ * Constant arc-length trajectories (not used here) are trajectories where the distance
+ * travelled in each segment in constant.
+ *
+ * @param constant_period_trajectory [out] A constant interpolation period trajectory
+ * generated by the specified path parameters.
+ *
+ * @param path_parameters [in/out] The path parameters that define the trajectory. The
+ * input path_parameters.num_elements defines the number of constant-parameterization
+ * elements, and the returned path_parameters.num_elements defines the number of
+ * constant-period elements in the trajectory,
  *
- *  TODO: Remove when #1322 is merged.
+ * @param interpolation_period  [in] The duration of time between successive trajectory
+ * elements. In seconds.
  *
- * @param forwards_trajectory This is the trajectory that will be modified in place to
- * become a reverse trajectory
+ * @return A status indicating whether or not generation was successful
  */
-void static app_trajectory_planner_reversePositionTrajectoryDirection(
-    PositionTrajectory_t* forwards_trajectory);
+TrajectoryPlannerGenerationStatus_t
+app_trajectory_planner_generateConstantInterpolationPeriodPositionTrajectory(
+    PositionTrajectory_t* constant_period_trajectory,
+    FirmwareRobotPathParameters_t* path_parameters, float interpolation_period);
 
 /**
- * This function generates a velocity trajectory that corresponds to the time-optimal
- * velocity to follow a specified path. This profile is based on the input position
- * trajectory
- *
- * @param position_trajectory [in] The position trajectory to build the velocity
- * trajectory from
- * @param velocity_trajectory [out] The velocity trajectory that corresponds to the
- * time-optimal velocity to follow a specified path
+ * Function generates a velocity trajectory composed of X/Y/angular velocities and their
+ * time dependence.
+ *
+ * @param position_trajectory [in] A completely defined position trajectory.
+ *
+ * @param velocity_trajectory [out] A completely defined velocity trajectory that has
+ * equally time-spaced elements. The velocity trajectory has the name number of elements
+ * as the input position trajectory.
+ *
+ * @param num_elements [in] The number of elements in the trajectory.
  */
 void app_trajectory_planner_generateVelocityTrajectory(
-    PositionTrajectory_t* position_trajectory, VelocityTrajectory_t* velocity_trajectory);
+    PositionTrajectory_t* position_trajectory, VelocityTrajectory_t* velocity_trajectory,
+    unsigned int num_elements);
```

Comment:
In this case, the velocity trajectory is built from the position one. It will have whatever path characteristics that the original position trajectory had